### PR TITLE
Workflow for GTS snow DA (Ready to merge)

### PR DESCRIPTION
**Pull request by jiaruidong2017**
All the changes were made in the branch `gts_snow`: 
  1. Modified the `submit_cycle.sh` to include the workflow for GTS snow DA. 
  2. Add the script `template.ufs-noahMP.namelist.gdas_gts` for FV3GDAS four-cycle runs. 
  3. Modified the script `template.coupler.res` for handling the snow DA in the operational 4 cycles.

### CODE REVIEW

**Comment by ClaraDraper-NOAA** [comment]
@jiaruidong2017 don't forget to add a description to your PR. You also need to put the "DRAFT: do not merge" label on it, so that I know not to merge it.

File: submit_cycle.sh [hunk 1]
```diff
@@ -39,21 +39,24 @@ export do_DA=YES  # do full DA update
...
-export OUTDIR=${CYCLEDIR}/exp_out/${exp_name}/output/      # directory where output will be saved
-ICSDIR="/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/offline_ICS/single/" # OUTDIR for experiment with initial conditions
+export WORKDIR=/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/workflow/experiment/workdir # temporary work dir 
+export OUTDIR=/scratch1/NCEPDEV/global/Jiarui.Dong/JEDI/workflow/experiment/output      # directory where output will be saved
+export ICSDIR=$OUTDIR
+#ICSDIR="/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/offline_ICS/single/" # OUTDIR for experiment with initial conditions
                                                            # will use ensemble of restarts if present, otherwise will try 
                                                            # to copy a non-ensemble restart into each ensemble restart
-                                
+
```
Comment (ClaraDraper-NOAA) [inline submit_cycle.sh]: Please change all variables above this line back to my values.

Comment (ClaraDraper-NOAA) [reply submit_cycle.sh]: And set CYCHR to 24

File: submit_cycle.sh [hunk 2]
```diff
@@ -386,6 +402,7 @@ if [ $THISDATE -lt $ENDDATE ]; then
     echo "export STARTDATE=${THISDATE}" > ${analdate}
     echo "export ENDDATE=${ENDDATE}" >> ${analdate}
     cd ${CYCLEDIR}
+#    rm -rf ${WORKDIR}
```
Comment (ClaraDraper-NOAA) [inline submit_cycle.sh]: delete this comment.

File: template.ufs-noahMP.namelist.gdas_gts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+&run_setup
```
Comment (ClaraDraper-NOAA) [inline template.ufs-noahMP.namelist.gdas_gts]: Delete this file, and use the template.ufs-noahMP.namelist.gdas one, and add code to sub in the model time step as is done for the template.coupler.res file. Also, add the same code to the other forcing files (*gefs and *gswp3)

**Review by ClaraDraper-NOAA — CHANGES_REQUESTED** [review]

File: submit_cycle.sh (hunk 1 above)
Comment (jiaruidong2017) [reply submit_cycle.sh]: Done

File: submit_cycle.sh (hunk 2 above)
Comment (jiaruidong2017) [reply submit_cycle.sh]: Deleted

File: template.ufs-noahMP.namelist.gdas_gts (hunk 3 above)
Comment (jiaruidong2017) [reply template.ufs-noahMP.namelist.gdas_gts]: Done

File: submit_cycle.sh [hunk 4]
```diff
@@ -29,7 +29,7 @@ export ensemble_size=1 # ensemble_size of 1 = do not run ensemble
 
 atmos_forc='gdas' # options: gdas, gswp3, gefs_ens
 
-dates_per_job=2 # number of cycles to submit in a single job
+dates_per_job=80 # number of cycles to submit in a single job
```
Comment (ClaraDraper-NOAA) [inline submit_cycle.sh]: Please change this back to 2.

**Review by ClaraDraper-NOAA — COMMENTED** [review]
Thanks Jiarui - this looks good. If you can make that one small change, I'll merge it.

File: submit_cycle.sh (hunk 4 above)
Comment (jiaruidong2017) [reply submit_cycle.sh]: Done

**Review by ClaraDraper-NOAA — APPROVED** [review]

**Pull request merged**